solutions/usaco-811.mdx

```diff
@@ -2,16 +2,87 @@
 id: usaco-811
 source: USACO Silver 2018 February
 title: Snow Boots
-author: Qi Wang
+author: Qi Wang, Ryan Chou
 ---
 
-[Official Analysis](http://www.usaco.org/current/data/sol_snowboots_silver_feb18.html)
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_snowboots_silver_feb18.html)
 
+## Explanation
+
+Due to the low bounds on $N$ and $B$, we can run a [DFS (Depth First Search)](/silver/dfs) on every possible location and boot size Farmer John could be wearing.
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N^2B + NB^2)$
 <LanguageSection>
+
+<CPPSection>
+	
+```cpp
+#include <iostream>
+#include <cstdio>
+#include <algorithm>
+
+using namespace std;
+
+int n;
+int m;
+int depths[251];
+bool vis[251][251];
+pair<int, int> shoes[251];
```

Comment:
1. put in constant MAX_N
2. when declaring these arrays, use MAX_N + 1 (or be a normal person & use 0-indexing)
3. even better, use vectors